server/services/auth.ts

```diff
@@ -0,0 +1,38 @@
+import bcrypt from 'bcrypt';
+import { User } from '../models';
+import { CustomError } from '../utils';
+import { signupSchema } from '../validations/schema';
+import { generateToken } from '../utils/jwt/generateToken';
+import { UserData } from '../interfaces/auth';
+
+const signupService = async (userData: UserData) => {
+  const { username, email, password, phone, role } = userData;
+
+  const validatedData = await signupSchema.validateAsync(userData);
+
+  if (!validatedData) {
```

Comment:
the ```validateAsync ``` throw Error without this choking 
and we will find it on the catch of errorWrapper